build-logic-commons/code-quality/build.gradle.kts

```diff
@@ -12,6 +12,8 @@ java {
 group = "gradlebuild"
 
 dependencies {
+    implementation("me.champeau.gradle:japicmp-gradle-plugin:0.3.0")
```

Comment:
Can we remove this dependency by making the plugin extensible? For example you could register custom rules to extract report locations to the root plugin extension. That way in the main build we register the things for the japicmp plugin and the ktlint plugin and we wouldn't need these dependencies here. Having the dependencies here also would mean we'd need to update them in two locations, right?

Reply:
Removed.